File: backend/app/schemas/document.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class DocumentResponse(BaseModel):
    id: UUID
    owner_id: UUID
    original_filename: str
    content_type: str
    file_size: int
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class DocumentDeleteResponse(BaseModel):
    message: str
    document_id: UUID


class DocumentProcessResponse(BaseModel):
    message: str
    document_id: UUID
    status: str
    character_count: int
    

class DocumentChunkResponse(BaseModel):
    id: UUID
    document_id: UUID
    chunk_index: int
    content: str
    character_count: int
    start_character: int
    end_character: int
    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )


class DocumentChunkingResponse(BaseModel):
    message: str
    document_id: UUID
    status: str
    chunk_count: int
    
class DocumentEmbeddingResponse(BaseModel):
    message: str
    document_id: UUID
    status: str
    embedded_chunk_count: int
    
class DocumentPrepareResponse(BaseModel):
    message: str
    document_id: UUID
    status: str
    character_count: int
    chunk_count: int
    embedded_chunk_count: int
    
class DocumentPreparationStartedResponse(BaseModel):
    message: str
    document_id: UUID
    status: str

File: frontend/src/types/document.ts
export type Document = {
  id: string;
  owner_id: string;
  original_filename: string;
  content_type: string;
  file_size: number;
  status: string;
  created_at: string;
};

export type DocumentChunk = {
  id: string;
  document_id: string;
  chunk_index: number;
  content: string;
  character_count: number;
  start_character: number;
  end_character: number;
  created_at: string;
};

export type DocumentDeleteResponse = {
  message: string;
  document_id: string;
};

export type DocumentProcessResponse = {
  message: string;
  document_id: string;
  status: string;
  character_count: number;
};

export type DocumentChunkingResponse = {
  message: string;
  document_id: string;
  status: string;
  chunk_count: number;
};

export type DocumentPrepareResponse = {
  message: string;
  document_id: string;
  status: string;
};